multi-arch-builders/builder-splunk.bu

```diff
@@ -0,0 +1,123 @@
+# This butane config will do the following:
+#
+# - create a splunk user
+# - build and start a splunk forwarder container to send journald logs to splunk
+#
+# See the rhcos multiarch builder splunk documentation in the CoreOS Team Operations GitLab repo.
+# The following env variables need to be set before creating the ignition config:
+#
+# - ${DEPLOYMENT_CLIENT}
+# - ${SPLUNK_HOSTNAME}
+# - ${ITPAAS_SPLUNK_REPO}
+# - ${SPLUNK_SIDECAR_REPO}
+#
+variant: fcos
+version: 1.4.0
+passwd:
+  users:
+    - name: splunk
+      groups:
+        - systemd-journal
+storage:
+  directories:
+    - path: /home/splunk/.config
+      user:
+        name: splunk
+      group:
+        name: splunk
+    - path: /home/splunk/.config/systemd
+      user:
+        name: splunk
+      group:
+        name: splunk
+    - path: /home/splunk/.config/systemd/user
+      user:
+        name: splunk
+      group:
+        name: splunk
+    - path: /home/splunk/.config/systemd/user/default.target.wants
+      user:
+        name: splunk
+      group:
+        name: splunk
+    - path: /home/splunk/.config/containers
+      user:
+        name: splunk
+      group:
+        name: splunk
+    - path: /home/splunk/.config/containers/systemd
+      user:
+        name: splunk
+      group:
+        name: splunk
+  files:
+    - path: /var/lib/systemd/linger/splunk
+      mode: 0644
+    - path: /home/splunk/.config/containers/systemd/splunk.container
+      mode: 0644
+      user:
+        name: splunk
+      group:
+        name: splunk
+      contents:
+        inline: |
+          [Unit]
+          Description=Start the splunk container to send journald logs to splunk
+          After=build-splunk.service
+          [Container]
+          ContainerName=rhcos-multiarch-splunk
+          Image=localhost/splunkforwarder-rhcos:latest
+          # set the user to root to run the splunk process
+          # as the splunk user on the host
+          User=root
+          Volume=/var/log/journal:/var/log/journal:ro
+          Volume=/run/log/journal:/run/log/journal:ro
+          Volume=/etc/machine-id:/etc/machine-id:ro
+          # use the container host's network to properly establish
+          # a connection with the splunk server
+          Network=host
+          # use `--ipc=host` to use the container host's IPC namespace
+          # to properly establish a connection with the splunk server
```

Comment:
usually `ipc` is for processes to communicate with each other on the same host IIUC. Is there message passing being done back and forth between `journalctl` running in the container and some daemon on the host? or something else going on?

Reply:
I would imagine it has something to do with the splunk process itself needing to communicate with the server. I was advised by the splunk team to use it this way. We can chat more about this 1:1.

Reply:
Hmm. I tried googling around a bit but couldn't find anything. Can you confirm that if you don't have this setting it doesn't work?

Reply:
Yes, I tried it earlier today, and again just now to make sure. It doesn't work without `--ipc=host`. I can see a connection with splunk, but no logs are being sent to the server.

Reply:
ok maybe just add a "we don't know why this is needed but it doesn't work without it" to the comment.